plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/property/BuildConfigPropertySource.java

```diff
@@ -23,25 +23,23 @@
 import com.openshift3.client.model.build.IGitBuildSource;
 import com.openshift3.client.model.build.ISTIBuildStrategy;
 
-public class BuildConfigPropertySource extends DefaultResourcePropertySource {
+public class BuildConfigPropertySource extends ResourcePropertySource<IBuildConfig> {
 	
 	private static final String TRIGGERS = "Triggers";
 	private static final String IMAGE = "Image";
 	private static final String ENVIRONMENT_VARIABLES = "Environment Variables";
 	private static final String STRATEGY = "Strategy";
 	private static final String SOURCE = "Source";
-	private IBuildConfig config;
 	
 	public BuildConfigPropertySource(IBuildConfig resource) {
 		super(resource);
-		this.config = resource;
 	}
 
 	@Override
 	public IPropertyDescriptor[] getResourcePropertyDescriptors() {
 		List<IPropertyDescriptor> all = new ArrayList<IPropertyDescriptor>();
 		all.addAll(getBuildTriggerPropertyDescriptors());
-		switch(config.getBuildStrategy().getType()){
```

Comment:
I think there is always a chance, but that would mean only a few things:

1.  There is an error on the server-side since I believe there is validation.  BuildConfigs always need a build strategy otherwise they arent useful
2.  We don't recognize this type of strategy which would be a version issue and I think gets back to the ticket we logged to re-evaluate the use of enums

----- Original Message -----
> From: "André Dietisheim" <notifications@github.com>
> To: "jbosstools/jbosstools-openshift" <jbosstools-openshift@noreply.github.com>
> Cc: "Jeff Cantrill" <jcantril@redhat.com>
> Sent: Thursday, February 19, 2015 3:32:59 AM
> Subject: Re: [jbosstools-openshift] [JBIDE-19285] Display replication controller specific properties (#551)
> 
> >  	}
> >  
> >  	@Override
> >  	public IPropertyDescriptor[] getResourcePropertyDescriptors() {
> >  		List<IPropertyDescriptor> all = new ArrayList<IPropertyDescriptor>();
> >  		all.addAll(getBuildTriggerPropertyDescriptors());
> > -		switch(config.getBuildStrategy().getType()){
> 
> any chance config can and/or config.getBuildStrategy are null?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jbosstools/jbosstools-openshift/pull/551/files#r24971947